core/models/asset.py

```diff
@@ -301,6 +302,42 @@ class Meta:
         ordering = ['-id']
 
 
+class AssetAssignee(models.Model):
+    department = models.OneToOneField('Department', null=True, blank=True,
+                                      on_delete=models.CASCADE)
+    user = models.OneToOneField('User', null=True, blank=True,
+                                on_delete=models.CASCADE)
+
+    def __str__(self):
+        if self.department:
+            return str(self.department)
+        if self.user:
+            return str(self.user)
+        else:
+            return "No Assignee Here"
+
+    @property
```

Comment:
Creating `@property` methods on the **AssetAssignee**  model to resolve fields on the **User**/**Department** model works for now, though it's not scalable. A better approach (dare I say _DRY_ er) is to reference the respective fields on the **User**/**Department**  model themselves. All the same, let's keep it as is for now. We'll work on refactoring this at a later stage.